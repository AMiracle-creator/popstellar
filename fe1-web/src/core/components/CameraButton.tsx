import PropTypes from 'prop-types';
import React from 'react';
import { Image, ImageStyle, StyleSheet, View } from 'react-native';

import circularButtonStyles from '../styles/stylesheets/circularButtonStyles';
import PoPTouchableOpacity from './PoPTouchableOpacity';

const cameraImage = require('resources/img/ic_camera.png');

/**
 * Camera button that executes an onPress action given in props
 */

const styles = StyleSheet.create({
  icon: {
    width: 64,
    height: 64,
  } as ImageStyle,
});

function CameraButton({ action }: IPropTypes) {
  return (
    <View>
      <PoPTouchableOpacity style={circularButtonStyles.button} onPress={action}>
        <Image style={styles.icon} source={cameraImage} />
      </PoPTouchableOpacity>
    </View>
  );
}

const propTypes = {
  action: PropTypes.func.isRequired,
};
CameraButton.prototype = propTypes;

type IPropTypes = PropTypes.InferProps<typeof propTypes>;

export default CameraButton;
