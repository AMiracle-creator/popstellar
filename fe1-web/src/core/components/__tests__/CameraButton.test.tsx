import React from 'react';
import { render } from '@testing-library/react-native';
import CameraButton from 'core/components/CameraButton';

let action: Function;

beforeEach(() => {
  action = jest.fn();
});

describe('CameraButton', () => {
  it('renders correctly', () => {
    const { toJSON } = render(
      <CameraButton action={() => action} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
