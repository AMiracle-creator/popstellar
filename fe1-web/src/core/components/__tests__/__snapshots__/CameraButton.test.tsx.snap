// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CameraButton renders correctly 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "#2196F3",
        "borderRadius": 80,
        "height": 80,
        "justifyContent": "center",
        "opacity": 1,
        "width": 80,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "height": 64,
          "width": 64,
        }
      }
    />
  </View>
</View>
`;
